hw_7_classes.py

```diff
@@ -0,0 +1,233 @@
+import json
+import pprint
+import os
+import pathlib
+import fnmatch
+
+
+# Задача-1
+# У вас есть список(list) IP адресов. Вам необходимо создать
+# класс который будет иметь методы:
+# 1) Получить список IP адресов
+# 2) Получить список IP адресов в развернутом виде
+# (10.11.12.13 -> 13.12.11.10)
+# 3) Получить список IP адресов без первых октетов
+# (10.11.12.13 -> 11.12.13)
+# 4) Получить список последних октетов IP адресов
+# (10.11.12.13 -> 13)
+
+
+class IpAddress:
+
+    ip_list = ['10.11.12.13', '192.168.12.1', '147.22.243.33']
+
+    def __init__(self):
+        self._list = IpAddress.ip_list
+        self._reversed_addr = []
+        self._last_3_ocktets = []
+        self._last_ocktets = []
+
+    def get_ip_list(self):
+        return f"IP addresses: {self._list}"
+
+    def get_reversed_ip_addresses(self):
+        for address in self._list:
+            self._reversed_addr.append('.'.join(address.split('.')[::-1]))
+        return f"Reversed addresses list: {self._reversed_addr}"
+
+    def get_last_3_ocktets_list(self):
+        for address in self._list:
+            self._last_3_ocktets.append('.'.join(address.split('.')[1:]))
+        return f"Last 3 ocktets: {self._last_3_ocktets}"
+
+    def get_last_ocktets_list(self):
+        for address in self._list:
+            self._last_ocktets.append(address.split('.')[-1])
+        return f"Last ocktets: {self._last_ocktets}"
+
+
+print('\nTask 1. IP addresses class\n')
+a = IpAddress()
+print(a.get_ip_list())
+print(a.get_reversed_ip_addresses())
+print(a.get_last_3_ocktets_list())
+print(a.get_last_ocktets_list())
+
+
+# Задача-2
+# У вас несколько JSON файлов. В каждом из этих файлов есть
+# произвольная структура данных. Вам необходимо написать
+# класс который будет описывать работу с этими файлами, а
+# именно:
+# 1) Запись в файл
+# 2) Чтение из файла
+# 3) Объединение данных из файлов в новый файл
+# 4) Получить путь относительный путь к файлу
+# 5) Получить абсолютный путь к файлу
+
+class JsonHandler:
+
+    default_file = 'example_json_1.json'
+
+    def __init__(self, file):
+        if file:
```

Comment:
no sense in this check. You can't initialize your class without setting a file argument, so the file will have a value each time when we initialize this class

Reply:
`def __init__(self, file=None):`